grapher/mapCharts/AnnotationGenerator.ts

```diff
@@ -74,539 +46,482 @@ export function generateAnnotations(
     let anchor = false
-    const k1 = [point[0], point[1]]
-    const kk = true
-    if (kk) {
-        if (type == ExternalPositions.top) {
-            k1[0] = k1[0] - textWidth / 2
-        } else if (type == ExternalPositions.bottom) {
-            k1[0] = k1[0] - textWidth / 2
-            k1[1] = k1[1] + fontSize
-        } else if (type == ExternalPositions.left) {
-            k1[0] = k1[0] - textWidth
-            k1[1] = k1[1] + fontSize / 2
-        } else if (type == ExternalPositions.right) {
-            k1[1] = k1[1] + fontSize / 2
-        } else if (
-            type == ExternalPositions.bottomLeft ||
-            type == ExternalPositions.topLeft
-        ) {
-            k1[0] = k1[0] - textWidth
-            k1[1] = k1[1] + fontSize / 2
-        }
-        let u = 1
-        let fin = false
-        if (country.area / viewportScale > 0.5)
-            for (let g = 1; g <= 8; g++) externalIncrement(type, k1)
-        else anchor = true
-        while (u <= 2) {
-            let more = true
-            externalIncrement(type, k1)
-            const markerEnd = markerEndPosition(
-                type as ExternalPositions,
-                k1,
-                textWidth,
-                fontSize
-            )
-            for (const x of combinedData) {
-                const g1 = [x.bounds.bottomLeft.x, x.bounds.bottomLeft.y]
-                const g2 = [x.bounds.bottomRight.x, x.bounds.bottomRight.y]
-                const g3 = [x.bounds.topRight.x, x.bounds.topRight.y]
-                const g4 = [x.bounds.topLeft.x, x.bounds.topLeft.y]
-                const rectPoints = [g1, g2, g3, g4, g1]
-                const b1 = k1
-                const b2 = [k1[0] + textWidth, k1[1]]
-                const b3 = [k1[0] + textWidth, k1[1] - fontSize]
-                const b4 = [k1[0], k1[1] - fontSize]
-                const firstPoint = x.path
-                    .slice(1, x.path.indexOf("L"))
-                    .split(",")
-                const labelPoints = [b1, b2, b3, b4, b1]
-                if (
-                    insideCheck(
-                        [Number(firstPoint[0]), Number(firstPoint[1])],
-                        labelPoints
-                    )
-                ) {
-                    more = false
-                    break
+    const p = [point[0], point[1]]
+    if (type == ExternalDirections.top) {
+        p[0] = p[0] - textWidth / 2
+    } else if (type == ExternalDirections.bottom) {
+        p[0] = p[0] - textWidth / 2
+        p[1] = p[1] + fontSize
+    } else if (type == ExternalDirections.left) {
+        p[0] = p[0] - textWidth
+        p[1] = p[1] + fontSize / 2
+    } else if (type == ExternalDirections.right) {
+        p[1] = p[1] + fontSize / 2
+    } else if (
+        type == ExternalDirections.bottomLeft ||
+        type == ExternalDirections.topLeft
+    ) {
+        p[0] = p[0] - textWidth
+        p[1] = p[1] + fontSize / 2
+    }
+    let j = 1
+    let check = true
```

Comment:
nit: like in the other function, it would be nice to give `check` a more specific name